data/src/main/java/me/androidbox/data/worker_manager/UploadEventWorker.kt

```diff
@@ -0,0 +1,100 @@
+package me.androidbox.data.worker_manager
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.Data
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import me.androidbox.data.R
+import me.androidbox.data.remote.model.response.EventDto
+import me.androidbox.data.remote.model.response.PhotoDto
+import me.androidbox.data.remote.network.event.EventService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.UploadEventImp.Companion.ERROR
+import me.androidbox.data.worker_manager.UploadEventImp.Companion.EVENT_PHOTOS
+import me.androidbox.data.worker_manager.util.CreatePhotoMultipart
+import okhttp3.MultipartBody
+
+@HiltWorker
+class UploadEventWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val eventService: EventService,
+    private val createPhotoMultipart: CreatePhotoMultipart,
+    private val moshi: Moshi
+) : CoroutineWorker(context, workerParameters)  {
+
+    override suspend fun doWork(): Result {
+        val listOfPhoto = inputData.getStringArray(UploadEventImp.EVENT_PHOTOS)?.toList()
+        val eventRequestJson = inputData.getString(UploadEventImp.EVENT)
+        val isEditMode = inputData.getBoolean(UploadEventImp.IS_EDIT_MODE, false)
+
+        val formData = getFormData(isEditMode)
+
+        if(!eventRequestJson.isNullOrEmpty()) {
+            val listOfPhotoMultiPart = listOfPhoto?.let {
+                /* Create the multipart for the list of photos */
+                createPhotoMultipart.createMultipartPhotos(listOfPhoto)
+            } ?: listOf()
+
+            val responseResult = checkResult {
+                eventService.createEvent(
+                    listOfPhoto = listOfPhotoMultiPart,
+                    eventBody = MultipartBody.Part.createFormData(formData, eventRequestJson)
+                )
+            }
+
+            val result = responseResult.fold(
+                onSuccess = { eventDto ->
+                    if(eventDto.photos.isNotEmpty()) {
+                        val outputData = photoToJson(eventDto)
+
+                        Result.success(outputData)
+                    }
+                    else {
+                        Result.success()
+                    }
+                },
+                onFailure = { throwable ->
+                    val errorDate = workDataOf(ERROR to throwable)
```

Comment:
Why is it named
errorDate?

Reply:
@MKiperszmid I named it errorData as I will create a workDataOf to show the error if the upload failed. What do you think this should be called?

Reply:
Yes, but there's a typo, you called it `errorDate`

Reply:
Should be errorData